include/proxy/v4/proxy.h

```diff
@@ -0,0 +1,2381 @@
+  struct accessor {
+    friend const std::type_info& proxy_typeid(
+        const adl_accessor_arg_t<F, IsDirect>& self) noexcept {
+      const proxy<F>& p = access_proxy<F>(self);
+      if (!p.has_value()) [[unlikely]] { return typeid(void); }
+      const proxy_typeid_reflector& refl = proxy_reflect<IsDirect, R>(p);
+      return *refl.info;
+    }
+___PRO4_DEBUG(
+    accessor() noexcept { std::ignore = &_symbol_guard; }
+
+   private:
+    static inline const std::type_info& _symbol_guard(
+        const adl_accessor_arg_t<F, IsDirect>& self) noexcept
+        { return proxy_typeid(self); }
+)
+  };
+
+  const std::type_info* info;
+};
+#endif  // __cpp_rtti >= 199711L
+
+template <std::size_t N>
+struct sign {
+  consteval sign(const char (&str)[N])
+      { for (std::size_t i = 0; i < N; ++i) { value[i] = str[i]; } }
+
+  char value[N];
+};
+template <std::size_t N>
+sign(const char (&str)[N]) -> sign<N>;
+
+struct wildcard {
+  wildcard() = delete;
+
+  template <class T>
+  [[noreturn]] operator T() const {
+#if __cpp_lib_unreachable >= 202202L
+    std::unreachable();
+#else
+    std::abort();
+#endif  // __cpp_lib_unreachable >= 202202L
+  }
+};
+
+}  // namespace details
+
+namespace skills {
+
+#if __STDC_HOSTED__ && __has_include(<format>)
+template <class FB>
+using format = typename FB::template add_convention<
+    details::format_dispatch, details::format_overload_t<char>>;
+
+template <class FB>
+using wformat = typename FB::template add_convention<
+    details::format_dispatch, details::format_overload_t<wchar_t>>;
+#endif  // __STDC_HOSTED__ && __has_include(<format>)
+
+#if __cpp_rtti >= 199711L
+template <class FB>
+using indirect_rtti = typename FB
+    ::template add_indirect_convention<details::proxy_cast_dispatch,
+        void(details::proxy_cast_context) &,
+        void(details::proxy_cast_context) const&,
+        void(details::proxy_cast_context) &&>
+    ::template add_indirect_reflection<details::proxy_typeid_reflector>;
+
+template <class FB>
+using direct_rtti = typename FB
+    ::template add_direct_convention<details::proxy_cast_dispatch,
+        void(details::proxy_cast_context) &,
+        void(details::proxy_cast_context) const&,
+        void(details::proxy_cast_context) &&>
+    ::template add_direct_reflection<details::proxy_typeid_reflector>;
+
+template <class FB>
+using rtti = indirect_rtti<FB>;
+#endif  // __cpp_rtti >= 199711L
+
+template <class FB>
+using as_view = typename FB
+    ::template add_direct_convention<details::view_conversion_dispatch,
+        facade_aware_overload_t<details::view_conversion_overload>>;
+
+template <class FB>
+using as_weak = typename FB
+    ::template add_direct_convention<details::weak_conversion_dispatch,
+        facade_aware_overload_t<details::weak_conversion_overload>>;
+
+}  // namespace skills
+
+template <details::sign Sign, bool Rhs = false>
+struct operator_dispatch;
+
+#define ___PRO4_DEF_LHS_LEFT_OP_ACCESSOR(Q, SELF, ...) \
```

Comment:
Same. Since these macros will be undefined, suggest keeping their original names